apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/Trace/SimpleManagedTracerTest.cs

```diff
@@ -254,6 +256,203 @@ public void MultipleSpans()
+                var childOneId = spans[0].SpanId;
+                var childTwoId = spans[1].SpanId;
+                return spans.Count == 5 &&
+                    IsValidSpan(spans[0], "child-one", parentId: 0) &&
+                    IsValidSpan(spans[1], "child-two", parentId: 0) &&
+                    IsValidSpan(spans[2], "grandchild-one", parentId: childOneId) &&
+                    IsValidSpan(spans[3], "grandchild-two", parentId: childTwoId) &&
+                    IsValidSpan(spans[4], "root");
+            };
+            mockConsumer.Setup(c => c.Receive(Match.Create(rootMatcher)));
+
+            var childThreadsReleased = new ManualResetEventSlim(initialState: false);
+
+            Func<string, string, Task> op = async (childName, grandchildName) =>
+            {
+                childThreadsReleased.Wait();
+
+                using (tracer.StartSpan(childName))
+                {
+                    await Task.Delay(30);
+                    using (tracer.StartSpan(grandchildName))
+                    {
+                        await Task.Delay(30);
+                    }
+                    await Task.Delay(30);
+                }
+            };
+            var t1 = Task.Run(() => op("child-one", "grandchild-one").Wait());
+            var t2 = Task.Run(() => op("child-two", "grandchild-two").Wait());
+
+            using (tracer.StartSpan("root"))
+            {
+                childThreadsReleased.Set();
+                await Task.WhenAll(t1, t2);
+            }
+
+            mockConsumer.VerifyAll();
+        }
+
+        [Fact]
+        public async Task MultipleSpans_Threads_Started_During_Span()
+        {
+            var mockConsumer = new Mock<IConsumer<TraceProto>>();
+            var tracer = SimpleManagedTracer.Create(mockConsumer.Object, ProjectId, TraceId);
+
+            ulong rootId = 0;
+            Predicate<IEnumerable<TraceProto>> rootMatcher = (IEnumerable<TraceProto> t) =>
+            {
+                var spans = t.Single().Spans;
+                rootId = spans[0].SpanId;
+                return spans.Count == 1 && IsValidSpan(spans[0], "root");
+            };
+            mockConsumer.Setup(c => c.Receive(Match.Create(rootMatcher)));
+
+            var childThreadsReleased = new ManualResetEventSlim(initialState: false);
+
+            Func<string, string, Task> op = async (childName, grandchildName) =>
+            {
+                childThreadsReleased.Wait();
+
+                using (tracer.StartSpan(childName))
+                {
+                    await Task.Delay(30);
+                    using (tracer.StartSpan(grandchildName))
+                    {
+                        await Task.Delay(30);
+                    }
+                    await Task.Delay(30);
+                }
+            };
+
+            Task t1;
+            Task t2;
+            using (tracer.StartSpan("root"))
+            {
+                t1 = Task.Run(() => op("child-one", "grandchild-one").Wait());
```

Comment:
I wanted to test with child threads and I wanted to make sure the child spans were not started until we setup the 2nd matcher below. Will this happen as well with your proposal?